Portfolio: details popup window

khanh-devos opened the pull request:
- Add HTML, CSS part
- it is responsive with mobile view, desktop view.

khanh-devos committed:
remove hideMenu and showMenu, add popup window

- improve toggleHamburgerBtn() to use it for all anchors <a> in MenuPage
- add popup window.

khanh-devos committed:
add desktop view

khanh-devos committed:
Add javascript part
- create Object projects with keys: title, img, tech, description
- use projectID to name for each relevant button as an input

khanh-devos committed:
fix ESlint Stylelinter

Kanyorok committed:
Change the pixels for the buttons

khanh-devos committed:
fix ESlint and Stylelint

khanh-devos committed:
stop scolling while popup appears

koachcode commented on index.html:
```diff
@@ -65,6 +63,44 @@ <h2>I am a Software Engineer</h2>
 
     <section id="works">
+      <!-- popup window for each project -->
+      <div id="popup">
+        <div id="popup-content">
+          <div id="popup-title-X">
+            <h3>Project Title</h3>
+            <button id="popup-X-btn" onclick="hidePopup()">
+              <img src="./imgs/x.svg" alt="">
+            </button>
+          </div>
+
+          <div id="popup-skill-btns">
+            <ul class="nav-buttons">
+              <!-- <li><a href=""></a></li> -->
+              
+            </ul>
+          </div>
+
+          <div id="popup-info">
+            <img width="100%" src="./imgs/works-bg.png" alt="project-image">
+            
+            <div id="popup-info-details">
+              <p>Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essent</p>
+              
+              <div id="function-btns">
```
- @khanh-devos  from the project requirements you are to render the projects done dynamically with javascript, but you rendered from the HTML. You should have a project array that holds the project details and, your projects rendered from the project array.  This will help you add new projects easily in the future by just adding to the array.

```javascript
const projects = [{
    name: "Submarine game",
    description: "A Phaser3 platform game built as part of the Microverse JavaScript program.",
    image: "link_to_image.jpg",
    technologies: ["Phaser3", "JavaScript", "Jest"],
    link: "link_to_live_version.com",
}]
```
- Also Modal popup data is populated with the data from the projects array.

koachcode reviewed (changes requested):
Hi @khanh-devos, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
![you-are-near](https://media.giphy.com/media/UdaSUYClr92cjWpeVx/giphy.gif)

## Highlights

- No linter error :+1: 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Please check the requirements of this project phase.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

khanh-devos committed:
modify menu.js

khanh-devos committed:
add popup window for both mobile and desktop
- create mobile projects and desktop projects inside an object type
- initial load render all work box(card) by js

khanh-devos committed:
disable onclick on 1st desktop  work card

nahnahdev commented on js/menu.js:
```diff
@@ -1,30 +1,240 @@
+  const lis = project.tech.map((tech) => `<li><a href="">${tech}</a></li>`);
+
+  // setup the popup
+  return `<div id="popup">
+  <div id="popup-content">
+    <div id="popup-title-X">
+      <h3>${project.title}</h3>
+      <button id="popup-X-btn" onclick="hidePopup()">
+        <img src="./imgs/x.svg" alt="close button">
+      </button>
+    </div>
+
+    <div id="popup-skill-btns">
+      <ul class="nav-buttons">
+        ${lis.join('')}
+      </ul>
+    </div>
+
+    <div id="popup-info">
+      <img width="100%" src="${project.img}" alt="project-image">
+
+      <div id="popup-info-details">
+        <p>${project.description}</p>
+
+        <div id="function-btns">
+          <button class="btn-type-1">
+            See Live
+            <img width="20" src="./imgs/socialIcons/popupIcon.svg" alt="popupIcon">
+          </button>
+          </button>
+          <button class="btn-type-1">
+            See Source
+            <img width="20" src="./imgs/socialIcons/gitIcon.svg" alt="gitIcon"></button>
+        </div>
+      </div>
+    </div>
+  </div>
+</div>`;
+}
```
- [OPTIONAL ] Great job so far @khanh-devos Your project looks great 😉 However I noticed that the second popup is missing the title, the languages and the paragraph text. So kindly ensure that these elements appear on the popup as expected so as to provide users with a great user experience 😉 
- Your Design now
<img width="987" alt="Screen Shot 2023-05-18 at 2 22 18 PM" src="https://github.com/khanh-devos/MyPortfolio/assets/102699958/cab9492f-7abe-466c-bd81-255061dca465">

khanh-devos replied:
yeah...

nahnahdev reviewed (approved):
## Project Status: APPROVED :green_circle: 
Hi @khanh-devos , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
<img src="https://media0.giphy.com/media/BPJmthQ3YRwD6QqcVD/giphy.gif"/>

## Highlights
- All requested changes implemented 🔥 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

nahnahdev reviewed (approved):
## Project Status: APPROVED :green_circle: 
Hi @khanh-devos , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
<img src="https://media0.giphy.com/media/BPJmthQ3YRwD6QqcVD/giphy.gif"/>

## Highlights
- All requested changes implemented 🔥 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

khanh-devos committed:
add title & description for 1st desktop work card
